src/lib.rs

```diff
@@ -0,0 +1,198 @@
+extern crate regex;
+use regex::Regex;
+use std::error;
+use std::fmt;
+
+
+pub struct Siso {
+    route: Route,
+}
+
+impl Siso {
+    pub fn register(&mut self, path_nodes : &[PathNode], value: RouteValue) -> Result<bool, SisoError> {
+        if 0 == path_nodes.len() {
+            panic!("Need a least one node for registering a value...");
+        }
+        return self.route.register(path_nodes, value);
+    }
+
+    pub fn find(&self, path_nodes : &[PathNode]) -> Result<RouteValue, SisoError> {
+        if 0 == path_nodes.len() {
+            panic!("Need a least one node for finding a value...");
+        }
+        return self.route.find(path_nodes);
+    }
+
+    pub fn new() -> Siso {
+        Siso { route: Route::new(PathNode::PathPart(String::from("[root]")), None) }
+    }
+}
+
+#[derive(Debug)]
+struct Route {
+    node: PathNode,
+    routes: Vec<Route>,
+    value: RouteValue,
+}
+
+impl Route {
+    fn register(&mut self, path_nodes : &[PathNode], value: RouteValue) -> Result<bool, SisoError> {
+        let path_node = &path_nodes[0];
+        let mut index = Err("");
```

Comment:
An `Option` would be a better fit here than `Result`

Reply:
You mean for the index of for the return value ?

Reply:
Assumed it was the index which makes more sense